src/passes/OptimizeInstructions.cpp

```diff
@@ -162,8 +159,8 @@ struct OptimizeInstructions : public WalkerPass<PostWalker<OptimizeInstructions,
 
   Pass* create() override { return new OptimizeInstructions; }
 
-  OptimizeInstructions() {
-    ensureDatabase();
+  void prepareToRun(PassRunner* runner, Module* module) override {
+    static DatabaseEnsurer ensurer;
```

Comment:
this commit removes a hack and instead uses a static variable as suggested by @jfbastien in https://github.com/WebAssembly/binaryen/pull/694/files/c24b9f9f6af61abd9ef124837bee41cbba35a8f2#r78034581